code/workspaces/infrastructure-api/src/controllers/projectsController.js

```diff
@@ -0,0 +1,96 @@
+import { check, matchedData } from 'express-validator';
+import { service } from 'common';
+import projectsRepository from '../dataaccess/projectsRepository';
+
+async function listProjects(request, response, next) {
+  try {
+    const projects = await projectsRepository.getAll() || [];
+    response.send(projects);
+  } catch (error) {
+    next(new Error(`Error listing projects: ${error.message}`));
+  }
+}
+
+async function getProjectByKey(request, response, next) {
+  const { key } = matchedData(request);
+
+  try {
+    const project = await projectsRepository.getByKey(key);
+    if (project) response.send(project);
```

Comment:
I think I have a preference to avoid `if/else` blocks with no braces. I find this style a little harder to read.